includes/Dashboard/Templates/Settings.php

```diff
@@ -502,19 +502,54 @@ function insert_settings_info() {
             $default_locations = dokan_get_option( 'location', 'dokan_geolocation' );
 
             if ( ! is_array( $default_locations ) || empty( $default_locations ) ) {
-                $default_locations = array(
+                $default_locations = [
                     'latitude'  => '',
                     'longitude' => '',
                     'address'   => '',
-                );
+                ];
             }
 
-            $find_address      = ! empty( $post_data['find_address'] ) ? sanitize_text_field( $post_data['find_address'] ) : $default_locations['address'];
-            $default_location  = $default_locations['latitude'] . ',' . $default_locations['longitude'];
-            $location          = ! empty( $post_data['find_address'] ) ? sanitize_text_field( $post_data['location'] ) : $default_location;
+            $find_address     = ! empty( $_POST['find_address'] ) ? wc_clean( wp_unslash( $_POST['find_address'] ) ) : $default_locations['address'];
+            $default_location = $default_locations['latitude'] . ',' . $default_locations['longitude'];
+            $location         = ! empty( $_POST['find_address'] ) ? wc_clean( wp_unslash( $_POST['location'] ) ) : $default_location;
```

Comment:
Why are you checking `empty` on `$_POST['find_address']` whereas you are using the value of `$_POST['location']`?
And also, as others above said, `wc_clean` is not needed everywhere.